filtered-resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/filteredresource/repository/H2PersistenceContractTest.java

```diff
@@ -14,23 +14,35 @@
  * limitations under the License.
  */
 
-package uk.gov.gchq.palisade.contract.filteredresource.h2;
+package uk.gov.gchq.palisade.component.filteredresource.repository;
 
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
 import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
 
 import uk.gov.gchq.palisade.contract.filteredresource.ContractTestData;
-import uk.gov.gchq.palisade.service.filteredresource.FilteredResourceApplication;
+import uk.gov.gchq.palisade.service.filteredresource.config.ApplicationConfiguration;
+import uk.gov.gchq.palisade.service.filteredresource.config.AsyncConfiguration;
 import uk.gov.gchq.palisade.service.filteredresource.model.TopicOffsetMessage;
+import uk.gov.gchq.palisade.service.filteredresource.repository.JpaTokenOffsetPersistenceLayer;
 import uk.gov.gchq.palisade.service.filteredresource.repository.TokenOffsetRepository;
 import uk.gov.gchq.palisade.service.filteredresource.service.OffsetEventService;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-@SpringBootTest(classes = FilteredResourceApplication.class, webEnvironment = WebEnvironment.NONE)
+@DataJpaTest
+@ContextConfiguration(classes = {ApplicationConfiguration.class, AsyncConfiguration.class, JpaTokenOffsetPersistenceLayer.class})
+@EnableAutoConfiguration
+@EntityScan(basePackages = "uk.gov.gchq.palisade.service.filteredresource.domain")
+@EnableJpaRepositories(basePackages = "uk.gov.gchq.palisade.service.filteredresource.repository")
+@AutoConfigureTestDatabase(replace = Replace.NONE)
 @ActiveProfiles("h2")
 class H2PersistenceContractTest {
```

Comment:
Probably worth renaming this test class
```suggestion
class H2PersistenceComponentTest {
```